app/controllers/users/sessions_controller.rb

```diff
@@ -16,6 +16,23 @@ class Users::SessionsController < Devise::SessionsController
   #   super
   # end
 
+  # Singing in with authentication token (needed when signing in automatically
+  # from another website)
+  def auth_token_create
+    user = User.find_by_email(params[:user_email])
+    if user.authentication_token == params[:user_token][0..-2]
```

Comment:
And also, can you only trim the `params[:user_token]` if they end with trailing slash?
e.g.
```ruby
user_token = params[:user_token]
# Remove trailing slash if present
user_token.chop! if user_token.end_with?('/')
```